### PR TITLE
[TLT-5942][Bugfix] Align TAO 7.2 skill contracts with runtime

### PR DESCRIPTION
**Pull request by nnagrajrao**
### What changes are proposed in this pull request?

- Refresh the generated DINOv3 skill schemas and templates with complete LoRA and preservation configuration plus configurable loss-log cadence.
- Document all six frozen DINOv3 Visual ChangeNet backbones, gated Hugging Face access, local staging, and exact architecture/path matching.
- Add the `val_far` best-checkpoint recipe and define its FAR-at-100%-recall semantics, including the single-class `100.0` sentinel.
- Require explicit downstream Visual ChangeNet checkpoints and retain only the narrow `train.classify.loss` compatibility stub while the documented image baseline remains TAO 7.1; tao-pytorch #107 removes that requirement in TAO 7.2.
- Add executable release-contract and schema-drift coverage for these workflows.

### Why are the changes needed?

The TAO 7.2 skill contracts had drifted from the runtime and source-of-truth dataclasses. This caused incomplete DINOv3 specs, ambiguous Visual ChangeNet checkpoint retention, unavailable backbone guidance, and an unclear compatibility boundary between the pinned TAO 7.1 image and the TAO 7.2 runtime fix.

### Related issues

- [TLT-5942](https://jirasw.nvidia.com/browse/TLT-5942) / [NVBug 6642016](https://nvbugs.nvidia.com/bug/6642016) — expose DINOv3 loss-log cadence.
- [TLT-5937](https://jirasw.nvidia.com/browse/TLT-5937) / [NVBug 6642001](https://nvbugs.nvidia.com/bug/6642001) — retain the best Visual ChangeNet FAR checkpoint.
- [TLT-5932](https://jirasw.nvidia.com/browse/TLT-5932) / [NVBug 6641994](https://nvbugs.nvidia.com/bug/6641994) — document frozen DINOv3 and gated-weight staging.
- [TLT-5931](https://jirasw.nvidia.com/browse/TLT-5931) / [NVBug 6641991](https://nvbugs.nvidia.com/bug/6641991) — complete DINOv3 LoRA and preservation contracts.
- [TLT-5928](https://jirasw.nvidia.com/browse/TLT-5928) / [NVBug 6641956](https://nvbugs.nvidia.com/bug/6641956) — cover all six DINOv3 backbone variants.
- [TLT-5921](https://jirasw.nvidia.com/browse/TLT-5921) / [NVBug 6602769](https://nvbugs.nvidia.com/bug/6602769) — require explicit downstream checkpoints.
- [TLT-5925](https://jirasw.nvidia.com/browse/TLT-5925) / [NVBug 6604829](https://nvbugs.nvidia.com/bug/6604829) — documents the version boundary and preserves the current pinned-image handoff while tao-pytorch #107 removes the runtime requirement.

### Does this PR introduce _any_ user-facing change?

Yes. DINOv3 specs expose the supported TAO 7.2 controls, and Visual ChangeNet guidance now produces explicit, reproducible checkpoint and frozen-backbone handoffs.

### How was this patch tested?

- `git diff --check origin/main...HEAD`
- `pre-commit run --from-ref origin/main --to-ref HEAD --show-diff-on-failure`
- `./scripts/validate-skills.sh --all`
- Focused schema-drift, release-contract, and executable inference-handoff suite: 36 passed in the pinned TAO container.
- Full `scripts/tests` suite: 269 tests passed before the final panel-only documentation/test-assertion cleanup; the affected suite was rerun afterward.

### Was this patch authored or co-authored using generative AI tooling?

Generated-by: OpenAI Codex (GPT-5)

### Release note

Align TAO 7.2 DINOv3 and Visual ChangeNet skill contracts with the runtime, including LoRA/preservation fields, loss-log cadence, DINOv3 frozen backbones, FAR checkpoint retention, and explicit downstream checkpoints.

### Checklist

- [x] Commits include DCO sign-offs.
- [x] New and changed Python files include SPDX headers.
- [x] Local focused tests and skill validation pass.
- [x] User-facing reference material and templates are updated.
- [x] All related Jira tickets and NVBugs are linked.

### Notes for reviewers

Dependencies:

- Schema provenance: [tao-core #35](https://github.com/NVIDIA-TAO/tao-core/pull/35).
- Runtime behavior: [tao-pytorch #107](https://github.com/NVIDIA-TAO/tao-pytorch/pull/107).

The tao-core and tao-pytorch PRs may merge independently. This skill-bank PR must merge/backport last, after tao-core #35 is merged and the TAO 7.2 image pin includes tao-pytorch #107. If tao-core #35 changes before merge, regenerate and recheck the DINOv3 schemas.

### CODE REVIEW

File: skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py [hunk 1]
```diff
@@ -86,10 +86,9 @@ def _build_inference_spec(
 
     Strips train/evaluate/export blocks. Keeps model + dataset architecture
     verbatim so backbone, lighting layout, image size, difference module, and
-    concat type all match the checkpoint. Adds a ``train.classify.loss`` stub
-    because TAO's PL classifier rebuilds its criterion on load and asserts the
-    loss/difference_module pairing — without this stub, load_from_checkpoint
-    raises before inference ever starts.
+    concat type all match the checkpoint. Non-training actions derive their
+    checkpoint-compatible criterion from the model architecture, so no
```
Comment (vpraveen-nv) [inline skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py]: Hard runtime dependency: dropping the `train.classify.loss` stub only works on containers that include NVIDIA-TAO/tao-pytorch#107. On current released containers, `train.classify.loss` defaults to `contrastive`, so an inference spec without the `train` subtree crashes at criterion construction for learnable-difference checkpoints (`Contrastive loss only supports Euclidean distance module`) — exactly the failure the old troubleshooting entry documented. Please either (a) gate this on the first container tag that ships #107 and state the minimum TAO version in SKILL.md/prepare-for-inference.md, or (b) keep emitting the stub until that container is the skill's documented baseline. Merge order alone isn't enough — the skills run against pulled images, not source.

File: skills/applications/tao-run-deft-aoi/references/baseline_spec.yaml [hunk 2]
```diff
@@ -113,16 +113,18 @@ dataset:
       augment: True
     num_classes: 2
 evaluate:
-  checkpoint: "${results_dir}/train/changenet_classify.pth"
+  # The loop discovers a successful train artifact and passes its explicit
+  # container path. Never derive this from the action-local results_dir.
+  checkpoint: ''
   trt_engine: "${results_dir}/gen_trt_engine/changenet-classify.trt"
```
Comment (vpraveen-nv) [inline skills/applications/tao-run-deft-aoi/references/baseline_spec.yaml]: Same `${results_dir}` rebasing hazard the checkpoint fields were just emptied for: `evaluate.trt_engine`/`inference.trt_engine` still interpolate `${results_dir}/gen_trt_engine/...`, which rebases to the *current action's* results dir and won't point at the engine produced by `gen_trt_engine`. If the loop always overrides these, a comment saying so would help; otherwise they should be emptied for consistency.

File: skills/models/tao-train-dinov3/schemas/train.schema.json [hunk 3]
```diff
@@ -137,15 +150,15 @@
         "last_layer_learning_rate": {
           "freeze_steps": 1250,
           "max_decay_steps": 2500000,
-          "val_base": 7.07e-06,
-          "val_final": 1e-06,
+          "val_base": 0.00000707,
```
Comment (vpraveen-nv) [inline skills/models/tao-train-dinov3/schemas/train.schema.json]: The float-literal style changed throughout the regenerated schemas (`7.07e-06` → `0.00000707`, `1e-06` → `0.000001`). Values are identical after parsing, but Python's `json.dump` emits the scientific form, which suggests these files weren't written by `scripts/generate_dataclass_schemas.py` as-is. If they were hand-assembled or produced by a different serializer, the next honest regeneration will churn all of these lines again. How were they generated? Prefer committing the generator output verbatim.

**Comment by christinayyw** [comment]
## TAO Skill Execution Eval Summary

**Latest run:** [tao-sb-OSS-develop #235](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/235/) — `3404ae75` @ 2026-08-20 20:08 UTC

| Skill | Eval Case | Backend | Verdict | Report |
|---|---|---|---|---|
| `skills/applications/tao-run-deft-aoi` | — | `claude` | ✅ Pass (⚠️ w/ self-correction) | [Latest Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/235/artifact/jenkins-eval-reports/skills-applications-tao-run-deft-aoi-claude/skill-eval-report.html) |
| `skills/applications/tao-run-deft-aoi` | — | `codex` | ✅ Pass (⚠️ w/ self-correction) | [Latest Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/235/artifact/jenkins-eval-reports/skills-applications-tao-run-deft-aoi-codex/skill-eval-report.html) |
| `skills/models/tao-train-visual-changenet` | — | `claude` | ✅ Pass | [Latest Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/235/artifact/jenkins-eval-reports/skills-models-tao-train-visual-changenet-claude/skill-eval-report.html) |
| `skills/models/tao-train-visual-changenet` | — | `codex` | ❓ Error | [Latest Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/235/artifact/jenkins-eval-reports/skills-models-tao-train-visual-changenet-codex/skill-eval-report.html) |


<sub>✅ Pass = expectation met without self-correction &nbsp;·&nbsp; ✅ Pass (⚠️ w/ self-correction) = expectation met but the agent rewrote / retried &nbsp;·&nbsp; ❌ Fail = expectation not met &nbsp;·&nbsp; ❓ Error = shard didn't produce an eval report</sub>

**Review by vpraveen-nv — APPROVED** [review]
minor nits. approved.

File: skills/models/tao-train-dinov3/schemas/train.schema.json (hunk 3 above)
Comment (nnagrajrao) [reply skills/models/tao-train-dinov3/schemas/train.schema.json]: Addressed in 1c0298d. Regenerated all four schemas from tao-core #35 with scripts/generate_dataclass_schemas.py and committed the generator JSON verbatim, including scientific float notation. The drift test now compares each complete schema, normalizing only generator metadata sourced from Python sets.

File: skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py (hunk 1 above)
Comment (nnagrajrao) [reply skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py]: Addressed in 1c0298d. I retained the minimal train.classify.loss stub while the skill baseline is pinned to TAO 7.1, documented that it must match the training spec, and identified tao-pytorch #107 in TAO 7.2 as the removal boundary. The executable handoff test now asserts the copied loss.

File: skills/applications/tao-run-deft-aoi/references/baseline_spec.yaml (hunk 2 above)
Comment (nnagrajrao) [reply skills/applications/tao-run-deft-aoi/references/baseline_spec.yaml]: Addressed in 1c0298d. Both evaluate.trt_engine and inference.trt_engine are now empty explicit inputs, matching the checkpoint handoff rule, and the release-contract test guards both fields.

**Comment by nnagrajrao** [comment]
@vpraveen-nv The updated head changes watched NVSkills paths. The svc-nvskills-signing write invitation to my fork is still pending acceptance; after it is accepted, could an upstream maintainer please comment `/nvskills-ci`? The dispatcher requires maintain/admin permission on the upstream repository.

**Comment by christinayyw** [comment]
## TAO Skill Execution Eval Summary

**Latest run:** [tao-sb-OSS-develop #238](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/238/) — `1c0298dc` @ 2026-08-20 21:37 UTC

| Skill | Eval Case | Backend | Verdict | Report |
|---|---|---|---|---|
| `skills/applications/tao-run-deft-aoi` | — | `claude` | ✅ Pass (⚠️ w/ self-correction) | [Latest Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/238/artifact/jenkins-eval-reports/skills-applications-tao-run-deft-aoi-claude/skill-eval-report.html)<br><details><summary>Previous Reports</summary>[Build #235 Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/235/artifact/jenkins-eval-reports/skills-applications-tao-run-deft-aoi-claude/skill-eval-report.html)</details> |
| `skills/applications/tao-run-deft-aoi` | — | `codex` | ❓ Error | [Latest Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/238/artifact/jenkins-eval-reports/skills-applications-tao-run-deft-aoi-codex/skill-eval-report.html)<br><details><summary>Previous Reports</summary>[Build #235 Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/235/artifact/jenkins-eval-reports/skills-applications-tao-run-deft-aoi-codex/skill-eval-report.html)</details> |


<sub>✅ Pass = expectation met without self-correction &nbsp;·&nbsp; ✅ Pass (⚠️ w/ self-correction) = expectation met but the agent rewrote / retried &nbsp;·&nbsp; ❌ Fail = expectation not met &nbsp;·&nbsp; ❓ Error = shard didn't produce an eval report</sub>
